### DISCUSSION TITLE
Nonlinear index of refraction

### DISCUSSION THREAD

**Discussion by IoannisTazes** [epochpic/epoch#234] [Q&A]
Dear developers 
Could you let me know how does the EPOCH code calculates the nonlinear index of refraction of a medium? in particular, does it take into account nonlinear effects such as Kerr-induced self-focussing? 
Thank you in advance

**Comment by TomGoffrey** [comment]
@IoannisTazes I've converted this to a Q&A discussion.

EPOCH doesn't calculate the refractive index of a medium. The laser in EPOCH is implemented as a boundary condition for the EM fields.

**Reply by IoannisTazes** [reply]
@TomGoffrey Thank you for your response!

It is not clear to me, does EPOCH take into account the self-focusing of the pulse due to the Kerr effect? since there is not refractive index in EPOCH, what is the method to introduce the self-focusing of the pulse caused by kerr lencing?

**Reply by mdhabibi** [reply]
I am also looking forward to hearing their response!
